utils/zerovec/src/varzerovec/mod.rs

```diff
@@ -0,0 +1,417 @@
+        }
+    }
+
+    /// Convert this into a mutable vector of the owned `T` type, cloning if necessary.
+    ///
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use std::str::Utf8Error;
+    /// # use zerovec::VarZeroVecError;
+    /// # use zerovec::VarZeroVec;
+    ///
+    /// let strings = vec!["foo".to_owned(), "bar".to_owned(),
+    ///                    "baz".to_owned(), "quux".to_owned()];
+    /// let bytes = VarZeroVec::get_serializable_bytes(&strings);
+    /// let mut vec: VarZeroVec<String> = VarZeroVec::try_from_bytes(&bytes)?;
+    ///
+    /// assert_eq!(vec.len(), 4);
+    /// let mutvec = vec.make_mut();
+    /// mutvec.push("lorem ipsum".into());
+    /// mutvec[2] = "dolor sit".into();
+    /// assert_eq!(&vec[0], "foo");
+    /// assert_eq!(&vec[1], "bar");
+    /// assert_eq!(&vec[2], "dolor sit");
+    /// assert_eq!(&vec[3], "quux");
+    /// assert_eq!(&vec[4], "lorem ipsum");
+    /// # Ok::<(), VarZeroVecError<Utf8Error>>(())
+    /// ```
+    pub fn make_mut(&mut self) -> &mut Vec<T>
+    where
+        T: Clone,
+    {
+        match self.0 {
+            VarZeroVecInner::Owned(ref mut vec) => return vec,
+            VarZeroVecInner::Borrowed(components) => {
+                let vec = components.iter().map(T::from_unaligned).collect();
+                let new_self = VarZeroVecInner::Owned(vec).into();
+                *self = new_self;
+                // recursion is limited since we are guaranteed to hit the Owned branch
+                self.make_mut()
+            }
+        }
+    }
+
+    /// Obtain an owned `Vec<T>` out of this
+    pub fn to_owned(&self) -> Vec<T>
+    where
+        T: Clone,
+    {
+        let mut new = self.clone();
+        new.make_mut();
+        match new.0 {
+            VarZeroVecInner::Owned(vec) => vec,
+            _ => unreachable!(),
+        }
+    }
+
+    /// If this is borrowed, get the borrowed slice
+    #[cfg(feature = "serde")]
+    pub(crate) fn get_slice_for_borrowed(&self) -> Option<&'a [u8]> {
+        match self.0 {
+            VarZeroVecInner::Owned(..) => None,
+            VarZeroVecInner::Borrowed(b) => Some(b.entire_slice()),
+        }
+    }
+
+    /// For a slice of `T`, get a list of bytes that can be passed to
+    /// `try_from_bytes` to recoup the same data.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use std::str::Utf8Error;
+    /// # use zerovec::VarZeroVecError;
+    /// # use zerovec::VarZeroVec;
+    ///
+    /// let strings = vec!["foo".to_owned(), "bar".to_owned(), "baz".to_owned()];
+    /// let bytes = VarZeroVec::get_serializable_bytes(&strings);
+    ///
+    /// let mut borrowed: VarZeroVec<String> = VarZeroVec::try_from_bytes(&bytes)?;
+    /// assert_eq!(borrowed.to_owned(), strings);
+    ///
+    /// # Ok::<(), VarZeroVecError<Utf8Error>>(())
+    /// ```
+    ///
+    pub fn get_serializable_bytes(elements: &[T]) -> Vec<u8> {
```

Comment:
As suggested above: make this `from_aligned(&[T])` and then have a proper `.as_bytes()` function